chain/client/src/client.rs

```diff
@@ -1399,26 +1399,35 @@ impl Client {
     }
 
     /// Called asynchronously when the ShardsManager finishes processing a chunk.
+    ///
+    /// It panics in the non-debug mode if the persisting of the completed chunk fails.
```

Comment:
log_assert_fails panics in the debug mode only and otherwise it just prints the error message.
I'm not sure what you're aiming for but please either fix the comment or change log_assert to assert.

Reply:
Personally I prefer such methods to return errors and let the caller method decide if it's panic worthy.